### ISSUE TITLE
Restored repository

### ISSUE THREAD

**Issue by hansmbakker** [wind-rider/nuimo-mqtt-manager#1]
Hi @shaftoe,
Since Nuimo hack is over I published everything on GitHub again.

There is no DM system here so I used an issue. Please close it when you found this :)

Kind regards,
Hans

**Comment by shaftoe** [comment]
Hi @wind-rider ! Cool, thanks for it.
Anyway, I've been exploring the idea of using [Home Assistant](https://home-assistant.io/) as a hub for Nuimo too and it looks very promising so far. I'm just waiting for the official Nuimo SDK to support Python v3 (small changes needed to fully support all the actions), at that point I'll publish upstream the Nuimo integration, it's already available here as a working draft: https://github.com/shaftoe/home-assistant/commit/1b18a809df3405ff50e2aec60c315af4372d0c2c
HTH

**Comment by shaftoe** [comment]
PS apparently I've got no rights to close the issue, I have only a green 'Comment' button available

**Comment by hansmbakker** [comment]
Hi @shaftoe,
I was pleasantly surprised you started working on an integration I also wanted :)

Maybe home-assistant is a good "host" for the protocol I thought of - connect your Nuimo to home assistant and let home assistant distribute the events that come from Nuimo, besides directly being controlled itself?

The cool thing is that home assistant is supposed to be always running!

**Comment by shaftoe** [comment]
@wind-rider exactly! It also supports MQTT natively, i.e. you can either just make HA subscribe to a topic, or have an integration publish to a topic, or use the internal bus feature, or both. My idea was to fire events to the HA bus but I don't have any use case yet, so maybe publishing to an MQTT topic is better. What's your take on this?

**Comment by shaftoe** [comment]
Sorry, earlier I didn't find time to properly understand your proposal, now I'm going through it and I see you spent quite some time to document it, good job! I like a lot your proposal for Nuimo-App protocol, so I'm thinking of implementing it in my HomeAssistant component if you're ok with that. I.e., instead of using the built HA bus, firing events to the MQTT topics instead

**Comment by hansmbakker** [comment]
Hi,
sorry for my late response.
Thanks for your nice feedback! It would be really a compliment if you or somebody else would pick up my idea and integrate it in your HomeAssistant component! Maybe I can help, but Python is new for me so I can't promise much yet :)

My idea was mostly a concept - free to be implemented anywhere. I'd also be happy if Senic (@larsblumberg) would integrate it in the phone app or the desktop app (I saw some users requested it on [Senic's uservoice](https://heysenic.uservoice.com/forums/275802-application-requests/suggestions/8171331-mqtt-integration))...

When implementing the protocol there might still be needed some filtering to do, either on the "manager" side or on the "client app" side. This is because the ring is so precise that it generates a lot of small rotation increments. The network and MQTT can handle this surprisingly well, but turning the ring quickly might put an unnecessary load on the network. However the bottleneck in my setup was really Philips Hue. The lamps / Hue bridge cannot process the large amount of commands fast enough which results in a traffic jam of Hue commands. 

As for the architecture in HomeAssistant - I'm not too familiar with the internals of HA, I don't know what you mean with "the internal bus feature"..
If you would implement this in HA, I would imagine it as follows:
- The HA Nuimo component keeps track of connected Nuimos
- The HA Nuimo component has a few "internal apps" (for controlling the lights, media) that the user can swipe through (swiping down)
- The HA Nuimo component can keep track of "external (MQTT) apps".
  - External apps can register themselves over MQTT
  - the HA Nuimo component mixes them with the internal apps when the user swipes through the apps
- When an internal app is active, there is no need for MQTT traffic, HA can handle the events on its own (light, media)
  - (optionally the HA Nuimo component can always publish events to MQTT even when an internal app is active)
- When an external app is active, the HA Nuimo component publishes the Nuimo events on the respective app's MQTT topic
- The HA Nuimo component always listens for register/unregister/focus/icon messages on MQTT
- The HA Nuimo component keeps track of a list of apps per Nuimo - so that if you have a Nuimo in the living room and a Nuimo in the bedroom, they don't interfere (since having photoshop controls on your bedroom Nuimo is not useful)

**Comment by craigdent** [comment]
This repository still maintained...?
Is there a home assistant or HomeKit plug-in for nuimo using the hub they’ve released..?

Be awesome to integrate it into HomeKit and use to control other smart home devices

**Comment by hansmbakker** [comment]
@craigdent I’m sorry but I’m not maintaining this repository. I wrote it as a participation for the Nuimo hacking contest but unfortunately it didn’t really get traction after that.

I would like it if Senic would open up their Hub for plugins, though.
